http_check/check.py

```diff
@@ -224,17 +223,7 @@ def send_status_down(loginfo, message):
                 self.warning("Skipping SSL certificate validation for %s based on configuration"
                              % addr)
 
-            instance_proxy = self.proxies.copy()
-
-            # disable proxy if necessary
-            if skip_proxy:
-                instance_proxy.pop('http')
-                instance_proxy.pop('https')
-            elif self.proxies['no']:
-                for url in self.proxies['no'].replace(';', ',').split(","):
-                    if url in parsed_uri.netloc:
-                        instance_proxy.pop('http')
-                        instance_proxy.pop('https')
+            instance_proxy = config_proxy_skip(self.proxies.copy(), addr, skip_proxy=skip_proxy)
```

Comment:
also, could we consider putting all the logic in one single `get_instance_proxy(proxies, addr, skip_proxy)` function that would return the proxy settings to use for all the requests made by the check instance? This would avoid the call to `self.proxies['no'] = get_no_proxy_from_env()` in the `__init__` method.

Finally, I'm not fully aware of what the constraints on the proxy settings logic are, but since we set some proxy settings in `AgentCheck.__init__` (https://github.com/DataDog/dd-agent/blob/5.13.0/checks/__init__.py#L363-L379), it would make sense to me to have this function as a method of `AgentCheck`